R/ard_attributes.R

```diff
@@ -1,29 +1,48 @@
-#' ARD Attributes
+#' Add variable attributes to an ARD object
 #'
+#' @description
+#' S3 generic providing methods for the following classes:
+#'
+#' * `data.frame`: Adds variable attributes to an ARD data frame.
+#'
+#' @param data (`data.frame`)\cr
+#'   a data frame
+#' @param ... Arguments passed to other methods
+#'
+#' @return an ARD data frame of class 'card'
+#'
+#' @examples
+#' df <- dplyr::tibble(var1 = letters, var2 = LETTERS)
+#' attr(df$var1, "label") <- "Lowercase Letters"
+#'
+#' ard_attributes(df, variables = everything())
+#'
+#' @export
+#'
+ard_attributes <- function(data, ...) {
+  UseMethod("ard_attributes")
+}
+
+
 #' Add variable attributes to an ARD data frame.
 #' - The `label` attribute will be added for all columns, and when no label
 #'   is specified and no label has been set for a column using the `label=` argument,
 #'   the column name will be placed in the label statistic.
 #' - The `class` attribute will also be returned for all columns.
 #' - Any other attribute returned by `attributes()` will also be added, e.g. factor levels.
 #'
-#' @param data (`data.frame`)\cr
-#'   a data frame
+#' @rdname ard_attributes
+#' @param variables ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   variables to include
 #' @param label (named `list`)\cr
 #'   named list of variable labels, e.g. `list(cyl = "No. Cylinders")`.
 #'   Default is `NULL`
-#' @param variables ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
-#'   variables to include
 #'
-#' @return an ARD data frame of class 'card'
 #' @export
-#'
-#' @examples
-#' df <- dplyr::tibble(var1 = letters, var2 = LETTERS)
-#' attr(df$var1, "label") <- "Lowercase Letters"
-#'
-#' ard_attributes(df, variables = everything())
-ard_attributes <- function(data, variables = everything(), label = NULL) {
+ard_attributes.data.frame <- function(data,
+                                      variables = everything(),
+                                      label = NULL,
+                                      ...) {
   set_cli_abort_call()
 
   # check inputs ---------------------------------------------------------------
```

Comment:
There is a `check_data_frame()` call that we can remove since the s3 method ensures its class